### ISSUE TITLE
Composer installed Drupal in wrong directory?

### ISSUE THREAD

**Issue by PebbleSun** [composer/composer#6861]
Hi

I am trying to install Drupal 8 with Composer using the Github composer drupal-project found here:

https://github.com/drupal-composer/drupal-project

I want to install Drupal in the public_html directory found here...  /home/username/public_html/ so I did cd public_html directory and ran the script from the git drupal-project from there. There was just this warning below and a warning about not using sudo to download Composer.

Warning: Composer should be invoked via the CLI version of PHP, not the cgi-fcgi SAPI

All appeared fine until I looked inside the public_html directory and I found it was empty!

The installation went fine but it placed all my files in a different directory! It placed them here:

/opt/cpanel/composer/bin/project_name/vendor/drupal

Does anyone know why Composer did this? Is it to do with the warning above about the PHP version? Can I copy and paste my project_name directory into my public_html directory without consequences?

Many thanks

**Comment by ThomasBerends** [comment]
This is unlikely to be related to Composer.

What was the exact command that you executed to install Drupal?

**Comment by PebbleSun** [comment]
Hi Thomas 

Thanks for your reply, it's great to get some feedback on the problem. The exact lines on the terminal prior to attempting the install were:

[root@80 public_html]# ls
cgi-bin
[root@80 public_html]# composer create-project drupal-composer/drupal-project:8.x-dev directory_name --stability dev --no-interaction
Content-type: text/html; charset=UTF-8

Warning: Composer should be invoked via the CLI version of PHP, not the cgi-fcgi SAPI
Do not run Composer as root/super usre! See https://getcomposer.org/root for details
Installing drupal-composer/drupal-project etc....

Then as I said above I expected to find the project to be in public_html with the directory name of 'directory_name' but it wasn't there. It was in the place I identified above:

/opt/cpanel/composer/bin/project_name/vendor/drupal

Do you know if I made a mistake?

Many thanks

**Comment by alcohol** [comment]
`create-project` essentially does a `git clone` of the package you specified ([`drupal-project`](https://github.com/drupal-composer/drupal-project)). After the `git clone`, composer will switch to the target directory, and run `composer install`. If you look in the repo you specified, you will see that it contains a `composer.json` which lists `drupal/core` and other packages as its dependencies. Dependencies always get installed in `vendor/`.

For more information on how composer works, see https://getcomposer.org
For more details on how to install Drupal, check with the Drupal community

**Comment by PebbleSun** [comment]
Resolved.
I have resolved this issue and it was to do with permissions. I had run the drupal-project script as sudo which I assumed gave me full root privileges but it doesn't. When I re-ran the script in exactly the same way in the public_html directory using root the script ran perfectly and the project was set up in the directory exactly as I expected and according to various guides I have read. Incidentally the following warning did not occur on this occasion either.

Warning: Composer should be invoked via the CLI version of PHP, not the cgi-fcgi SAPI

Many thanks for your help with this issue.
